### ISSUE TITLE
Bitmart - cannot create order after upgrading to 4.3.76

### ISSUE THREAD

**Issue by krasnyd** [ccxt/ccxt#23354]
### Operating System

Windows

### Programming Languages

Python

### CCXT Version

4.3.76

### Description

I have tested the latest CCXT version with the updated futures Bitmart API from v1 to v2 (https://github.com/ccxt/ccxt/pull/23267), and my create order function stopped working. I am getting the error `{"code":40012,"message":"Invalid order"}`, but when I execute the same code with CCXT 4.3.58 it works without any problems. 

Is there some error on my side, or is there some change I need to make to work with their new API? The error code `40012` is marked in their documentation as `System error`, so can it be an error on the Bitmart side?

I have also tried to make a stop loss trigger order and failed with an error "Invalid leverage". I can provide my code if it would have helped. 

### Code

```python
import ccxt

coin = "ETCUSDT"
type = "market"
side = "sell"
opening_side_enum = 1 if side == "buy" else 4
leverage = 2
amount = 1

client = ccxt.bitmart({
    'apiKey': "...",
    'secret': "...",
    'uid': "...",
    "verbose": False
})

client.create_order(coin, type, side, amount, params={'leverage': leverage, 'marginMode': 'isolated', 'side': opening_side_enum})

positions = client.fetch_positions()
for position in positions:
    print("position", position)



```

**Comment by sc0Vu** [comment]
@krasnyd It seems not related to ccxt. I can create order. Do you have any other orders or positions?

```
$ p bitmart createOrder ETHUSDT market sell 1 None '{"leverage": 2, "marginMode": "isolated", "side": 4}'
Python v3.11.3
CCXT v4.3.76
bitmart.createOrder(ETHUSDT,market,sell,1,None,{'leverage': 2, 'marginMode': 'isolated', 'side': 4})
{'amount': 1,
 'average': None,
 'clientOrderId': None,
 'cost': None,
 'datetime': None,
 'fee': None,
 'fees': [],
 'filled': None,
 'id': '3000013071018888',
 'info': {'order_id': '3000013071018888', 'price': 'market price'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': 'sell',
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'ETH/USDT:USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'market'}
```

**Comment by krasnyd** [comment]
@sc0Vu Can you please try it on the ccxt version 4.3.76? It looks like you have version 4.3.74, which is before the mentioned pull request.

**Comment by carlosmiei** [comment]
Hello @krasnyd we recently migrated bitmart to v2, so that might be related to that. Can you post a verbose log? I will run a few tests on my end as well.

**Comment by carlosmiei** [comment]
@krasnyd The weird thing is, sometimes the exact same request fails first and then if I try again using the exact same parameters it works...

Example

first try
```
bitmart.createOrder (LTC/USDT:USDT, market, buy, 1)
fetch Request:
 bitmart POST https://api-cloud-v2.bitmart.com/contract/private/submit-order 

RequestBody:
 {"symbol":"LTCUSDT","type":"market","size":1,"side":1,"open_type":"cross"} 

handleRestResponse:
 bitmart POST https://api-cloud-v2.bitmart.com/contract/private/submit-order 200 OK 
ResponseHeaders:

ResponseBody:
 {"code":1000,"message":"Ok","data":{"order_id":3000013071021024,"price":"market price"},"trace":"5534867fdee4403aba27b34c6922da0f.125.17231147914900069"} 

2024-08-08T10:59:51.711Z iteration 0 passed in 625 ms

{
  id: '3000013071021024',
  info: { order_id: '3000013071021024', price: 'market price' },
  symbol: 'LTC/USDT:USDT',
  type: 'market',
  side: 'buy',
  amount: 1,
  trades: [],
  fees: []
}
```


second try:
```
CCXT v4.3.76
bitmart.createOrder (LTC/USDT:USDT, market, sell, 1)
fetch Request:
 bitmart POST https://api-cloud-v2.bitmart.com/contract/private/submit-order 
RequestHeaders:

RequestBody:
 {"symbol":"LTCUSDT","type":"market","size":1,"side":4,"open_type":"cross"} 

handleRestResponse:
 bitmart POST https://api-cloud-v2.bitmart.com/contract/private/submit-order 200 OK 

ResponseBody:
 {"code":1000,"message":"Ok","data":{"order_id":3000013071021030,"price":"market price"},"trace":"5534867fdee4403aba27b34c6922da0f.129.17231148700511785"} 

2024-08-08T11:01:10.260Z iteration 0 passed in 614 ms

{
  id: '3000013071021030',
  info: { order_id: '3000013071021030', price: 'market price' },
  symbol: 'LTC/USDT:USDT',
  type: 'market',
  side: 'sell',
  amount: 1,
  trades: [],
  fees: []
}
```

**Comment by carlosmiei** [comment]
@krasnyd Besides this inconsistent behavior from bitmart, we found indeed an issue with trigger orders, but will be fixed shortly here: https://github.com/ccxt/ccxt/pull/23367

**Comment by krasnyd** [comment]
Thank you for your reactions. I tested it today with the latest CCXT version (4.3.79), and I was able to create and close orders without any problems. I don't know if Bitmart made some fixes on their side, but everything seems to work on the v2, at least for me. 
I also discovered that I didn't have enough cash on the v2 account, so that could probably be the reason why I received the "Invalid order" error.
I made a few orders, but I didn't encounter the "random behavior", that @carlosmiei describes. 

I think this issue can be closed if you agree.
